explorer/src/lib/components/map/Map.svelte

```diff
@@ -0,0 +1,57 @@
+<script>
+  // @ts-nocheck
+
+  import { geoNaturalEarth1, geoPath } from "d3";
+  import dataset from "./world-map.json";
+
+  import "./Map.css";
+
+  /** @type {Array<{lat: number, lon:number}> | Error}*/
+  export let nodes;
+
+  /** @type {import("d3").GeoProjection}*/
+  const projection = geoNaturalEarth1();
+
+  /** @type {import("d3").GeoPath<any, import("d3").GeoPermissibleObjects>} */
+  const path = geoPath(projection);
+
+  /** @param {Array<{lat:number, lon:number}> | Error} data */
+  function removeDuplicates(data) {
+    if (Array.isArray(data)) {
+      return data.filter(
+        (value, index, self) =>
+          index ===
+          self.findIndex((t) => t.lat === value.lat && t.lon === value.lon)
+      );
+    } else {
+      return [];
+    }
+  }
```

Comment:
Just some suggestions, not a big deal, but I would refactor this a bit.

To start with, I would have the first `if` as

```
if (!Array.isArray(data)) {
 return [];
}
```

2) I would avoid using generic variable names, such as `t` – perhaps `location` would make more sense?

The `filter` bit itself could be rewritten to something like:

```
return data.filter((entry, index, self) => {
  // Find the first index of an entry with the same latitude and longitude
  const firstIndex = self.findIndex(
    (location) => location.lat === entry.lat && location.lon === entry.lon
  );
  // Keep the entry if its index is the first occurrence
  return index === firstIndex;
});
```

So the whole function can be something like:

<img width="639" alt="Screenshot 2024-05-07 at 15 17 17" src="https://github.com/dusk-network/rusk/assets/34094428/9794358c-1816-44da-8aad-dea2853cd612">